logrotate.c

```diff
@@ -2860,6 +2869,36 @@ static int readState(const char *stateFilename)
     return 0;
 }
 
+static int lockState(const char *stateFilename)
+{
+    int lockFd = open(stateFilename, O_RDWR);
+    if (lockFd == -1) {
+        if (errno == ENOENT) {
+            return 0;
```

Comment:
Does it mean we will proceed with rotations without holding any lock in this case?

Reply:
Now `lockState()` creates an empty stub statefile in that case